pkg/model/run.go

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright © 2019 One Concern
+ *
+ */
+
+package model
+
+import (
+	"time"
+)
+
+const (
+	Version = 1
+)
+
+type Category struct {
+	Name        string      `json:"name" yaml:"name"`               // Unique name for the category
+	Message     string      `json:"message" yaml:"message"`         // Message describing the category.
+	Contributor Contributor `json:"contributor" yaml:"contributor"` // Credentials for who created the category.
+	Version     int         `json:"version" yaml:"version"`         // Version of the run Schema.
+}
+
+type Run struct {
+	Name           string      `json:"name" yaml:"name"`                     // Name for the run
+	InstanceID     string      `json:"instanceid" yaml:"instanceid"`         // External ID used for a run. Not guaranteed to be unique.
+	RunID          string      `json:"id" yaml:"id"`                         // Internally generated Unique id for an instance of a run. Each time the same Type+Name is run it results in a new ID.
+	Timestamp      time.Time   `json:"timestamp" yaml:"timestamp"`           // Timestamp for when the run was created.
+	Contributor    Contributor `json:"contributor" yaml:"contributor"`       // Contributor who created the run.
+	RunFile        string      `json:"runfile" yaml:"runfile"`               // Raw file describing the run. Example: A yaml file.
```

Comment:
Would this be the argo workflow?

Reply:
For now yes.

Reply:
So it's a long string whose contents is the entire contents of the yaml?

Reply:
my understanding of `RunFile` is that it, together with `PipelineEngine` form a sort of [tagged union](https://en.wikipedia.org/wiki/Tagged_union) within the `Run` struct.

i kinda recommend making an interface type

```
interface RunEngine (
  RunFile() string
  PipelineEngine() string
)
```

and including a concrete type implementing only this interface in `Run` -- i.e.
```
type Run struct {
...
  Engine RunEngine
...
}
```
this'd allow inquiring `Run.Engine.(PipelinesTool)` or `Run.Engine.(Argo)`

Reply:
@galvare2 yes it would be the string content generated by the pipeline tool

Reply:
@ransomw1c for now splitting the runengine out as a type. The interface can be added later